lib/mongo/index/view.rb

```diff
@@ -251,6 +254,10 @@ def send_initial_query(server)
 
       def with_generated_names(models)
         models.dup.each do |model|
+          if model[:key].key?(:partial_filter_expression)
+            model[:partial_filter_expression] = model[:key][:partial_filter_expression]
+            model[:key].delete(:partial_filter_expression)
+          end
```

Comment:
No, it should be:

```ruby
client[:test].indexes.create_one({ "x" => 1}, partial_filter_expression: { "a" : { "$lte" => 1.5 }}})
```

The way you are doing it makes the options part of the index key, it needs to be clear that they are 2 separate parameters.